### PR TITLE
feat(clustering) add support for checking against OCSP responder found

### PR DESCRIPTION
**Pull request by dndx**

### CODE REVIEW

File: kong.conf.default [hunk 1]
```diff
@@ -278,6 +278,25 @@
                          # 14 days. That is, if CP haven't heard from a DP for
                          # 14 days, its entry will be removed.
 
+#cluster_ocsp = off
+                         # Whether to check for revocation status of DP
+                         # certificates using OCSP (Online Certificate Status Protocol).
```
Comment (fffonion) [inline kong.conf.default:283]: Let's mention the check is done by CP side, not DP is checking its own cert.

File: kong.conf.default [hunk 2]
```diff
@@ -278,6 +278,25 @@
...
+                         # If enabled, the DP certificate should contain the
+                         # "Certificate Authority Information Access" extension
+                         # and the OCSP method with URI of which the OCSP responder
+                         # can be reached.
```
Comment (fffonion) [inline kong.conf.default:287]:
```suggestion
                         # can be reached from CP.
```

File: kong/clustering.lua [hunk 3]
```diff
@@ -333,6 +336,66 @@ local function validate_shared_cert()
...
+
+    local ocsp_resp = res.body
+    if ocsp_resp and #ocsp_resp > 0 then
+      res, err = ocsp.validate_ocsp_response(ocsp_resp, der_cert)
+      if not res then
+        return false, "failed to validate OCSP response: " .. err
+      end
+    end
+
+    return true
```
Comment (fffonion) [inline kong/clustering.lua:394]: `if not ocsp_resp or #ocsp_resp == 0 `somehow, should we return false instead?

File: spec/02-integration/09-hybrid_mode/05-ocsp_spec.lua [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+local function set_ocsp_status(status)
+  local upstream_client = helpers.http_client(helpers.mock_upstream_host, helpers.mock_upstream_port, 5000)
+  local res = assert(upstream_client:get("/set_ocsp?status=" .. status))
+  assert.res_status(200, res)
+  upstream_client:close()
+end
+
+
+for _, strategy in helpers.each_strategy() do
+  describe("cluster_ocsp = on works with #" .. strategy .. " backend", function()
```
Comment (fffonion) [inline spec/02-integration/09-hybrid_mode/05-ocsp_spec.lua:18]: we can also add a `cluster_ocsp = on` + `OCSP responder errors, DP are disallowed`

**Review by fffonion — COMMENTED** [review]
Looks good overall, left some comments.

File: kong/clustering.lua (hunk 3 above)
Comment (dndx) [reply kong/clustering.lua:396]: Good catch!

**Review by fffonion — APPROVED** [review]

**Pull request merged**